test/parallel/test-whatwg-encoding-textdecoder.js

```diff
@@ -0,0 +1,94 @@
+// Flags: --expose-internals
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const { TextDecoder, TextEncoder } = require('util');
+const { customInspectSymbol: inspect } = require('internal/util');
+
+const buf = Buffer.from([0xef, 0xbb, 0xbf, 0x74, 0x65,
+                         0x73, 0x74, 0xe2, 0x82, 0xac]);
+
+// Make Sure TextDecoder exist
+assert(TextDecoder);
+
+// Test TextDecoder, UTF-8, fatal: false, ignoreBOM: false
+{
+  ['unicode-1-1-utf-8', 'utf8', 'utf-8'].forEach((i) => {
+    const dec = new TextDecoder(i);
+    const res = dec.decode(buf);
+    assert.strictEqual(res, 'test€');
+  });
+
+  ['unicode-1-1-utf-8', 'utf8', 'utf-8'].forEach((i) => {
+    const dec = new TextDecoder(i);
+    let res = '';
+    res += dec.decode(buf.slice(0, 8), { stream: true });
+    res += dec.decode(buf.slice(8));
+    assert.strictEqual(res, 'test€');
+  });
+}
+
+// Test TextDecoder, UTF-8, fatal: false, ignoreBOM: true
+{
+  ['unicode-1-1-utf-8', 'utf8', 'utf-8'].forEach((i) => {
+    const dec = new TextDecoder(i, { ignoreBOM: true });
+    const res = dec.decode(buf);
+    assert.strictEqual(res, '\ufefftest€');
+  });
+
+  ['unicode-1-1-utf-8', 'utf8', 'utf-8'].forEach((i) => {
+    const dec = new TextDecoder(i, { ignoreBOM: true });
+    let res = '';
+    res += dec.decode(buf.slice(0, 8), { stream: true });
+    res += dec.decode(buf.slice(8));
+    assert.strictEqual(res, '\ufefftest€');
+  });
+}
+
+// Test TextDecoder, UTF-8, fatal: true, ignoreBOM: false
+if (common.hasIntl) {
+  ['unicode-1-1-utf-8', 'utf8', 'utf-8'].forEach((i) => {
+    const dec = new TextDecoder(i, { fatal: true });
+    assert.throws(() => dec.decode(buf.slice(0, 8)),
+                  common.expectsError({
+                    code: 'ERR_ENCODING_INVALID_ENCODED_DATA',
+                    type: TypeError,
+                    message:
+                      /^The encoded data was not valid for encoding utf-8$/
```

Comment:
You could use a simple string, should be semantically the same as an anchored `RegExp`.